lib/ui/login/login_screen.dart

```diff
@@ -108,102 +122,129 @@ class _LoginScreenState extends State<LoginScreen>
             ),
           ),
         ),
-        child: Text(
-          AppLocalizations.of(context)!.login,
-          style: Theme.of(context).textTheme.labelSmall?.copyWith(
+        child: _isLoading
+            ? const CircularProgressIndicator(
                 color: AppColors.blackRussian,
+              )
+            : Text(
+                AppLocalizations.of(context)!.login,
+                style: Theme.of(context).textTheme.labelSmall?.copyWith(
+                      color: AppColors.blackRussian,
+                    ),
               ),
-        ),
       );
 
-  Stack _buildLoginScreen(BuildContext context) => Stack(
-        children: [
-          DimmedBackground(
-            background: Assets.images.splashBackground.path,
-            startOpacity: 0.01,
-            endOpacity: 1.0,
-            isAsset: true,
-          ),
-          FadeTransition(
-            opacity: _overlayOpacityAnimationController,
-            child: BackdropFilter(
-              filter: ImageFilter.blur(
-                sigmaX: 25.0,
-                sigmaY: 25.0,
-              ),
-              child: Container(
-                decoration: BoxDecoration(
-                  color: Colors.white.withOpacity(0.0),
-                ),
+  Stack _buildLoginScreen(
+      BuildContext context, bool isLoginSucceed, String errorMessage) {
+    if (isLoginSucceed) {
+      Future.delayed(Duration.zero, () async {
+        context.go("/home");
+      });
+    }
+    if (errorMessage.isNotEmpty) {
+      setState(() {
+        _isLoading = false;
+      });
+      showSnackBar(context, errorMessage);
+    }
```

Comment:
Please make use of `LoginState` to listen & execute the action when the state changes. For example:

```
  @override
  Widget build(BuildContext context) {
    ref.listen<LoginState>(loginViewModelProvider, (_, loginState) {
      loginState.maybeWhen(
        error: (error) {
          // show error
        },
        success: () async {
          // navigate
        },
        orElse: () {},
      );
    });
    return _buildLoginPage();
  }
```